serverless.yml

```diff
@@ -13,12 +13,31 @@ provider:
       Action:
         - "ses:SendEmail"
       Resource: "*"
+    - Effect: "Allow"
+      Action:
+        - "s3:CreateBucket"
+      Resource:
+        - arn:aws:s3:::${file(env.${opt:stage}.yml):S3_LOGGING_BUCKET}
     - Effect: "Allow"
       Action:
         - "s3:PutObject"
       Resource: 
         - arn:aws:s3:::${file(env.${opt:stage}.yml):S3_LOGGING_BUCKET}/*
 
+resources:
+  Conditions:
+    CreateS3Bucket: 
+      Fn::Not:
+        - Fn::Equals:
+          - ${file(env.${opt:stage}.yml):S3_LOGGING_BUCKET}
+          - ""
```

Comment:
This comes mostly from my ignorance of CloudFormation semantics, but does this bucket get removed during `sls deploy` or `sls remove`? If that happens, it might not be desirable to do this through serverless, since we probably want to keep the submission history even after the Lambdas have been upgraded/removed

Reply:
If you do `sls remove` and the bucket isn't empty, serverless will complain at you and it won't delete the stack. Similarly, it'll complain if you deploy, get letters sent and added to the bucket, then change the environment variable to point to a different bucket, then try to deploy again. You'd have to go into CloudFormation and manually delete the stack, opting to keep the bucket. CloudFormation won't ever delete a bucket that's got things in it, only if it's empty.